src/App.jsx

```diff
@@ -13,34 +15,135 @@ export const goodsFromServer = [
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (state === STATE_ALPHABETICALLY) {
+    if (stateArray.includes(STATE_REVERSE)) {
+      return listForSorting
+        .sort((item1, item2) => item1.localeCompare(item2))
+        .reverse();
+    }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    return listForSorting.sort((item1, item2) => item1.localeCompare(item2));
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (state === STATE_BY_LENGTH) {
+    if (stateArray.includes(STATE_REVERSE)) {
+      return listForSorting
+        .sort((item1, item2) => item1.length - item2.length)
+        .reverse();
+    }
+
+    return listForSorting.sort((item1, item2) => item1.length - item2.length);
+  }
+
+  if (state === STATE_REVERSE) {
+    return listForSorting.reverse();
+  }
+
+  return list;
+}
+
+function addStates(state, stateArray, prev) {
+  if (!stateArray.includes(state)) {
+    return [...prev, state];
+  }
+
+  if (stateArray.includes(state)) {
+    return stateArray.filter(item => item !== state);
+  }
+
+  return [...prev];
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [state, setState] = useState(null);
+  const [listOfGoods, setListOfGoods] = useState(goodsFromServer);
+  const [stateArray, setStateArray] = useState([]);
+  const listForPrint = sorting(listOfGoods, state, stateArray);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button is-info ', {
+            'is-light':
+              !stateArray.includes(STATE_ALPHABETICALLY) &&
+              (!stateArray.includes(STATE_ALPHABETICALLY) ||
+                !stateArray.includes(STATE_REVERSE)),
+          })}
+          onClick={() => {
+            setState(STATE_ALPHABETICALLY);
+            setStateArray(prev => {
+              return addStates(STATE_ALPHABETICALLY, stateArray, prev);
+            });
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button is-success ', {
+            'is-light':
+              !stateArray.includes(STATE_BY_LENGTH) &&
+              (!stateArray.includes(STATE_BY_LENGTH) ||
+                !stateArray.includes(STATE_REVERSE)),
+          })}
+          onClick={() => {
+            setState(STATE_BY_LENGTH);
+            setStateArray(prev => {
+              return addStates(STATE_BY_LENGTH, stateArray, prev);
+            });
+          }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**